File: src/dlp/ast/sentences/ConditionalSentence.ts

import {Sentence} from "./Sentence";
import {AbstractLocatable} from "../AbstractLocatable";
import {Expression} from "../expressions/Expression";

export abstract class ConditionalSentence extends AbstractLocatable implements Sentence{

    condition: Expression;
    body: Sentence;

    constructor(condition: Expression, body: Sentence, line: number, column: number) {
        super(line,column);
        this.condition = condition;
        this.body = body;
    }
}

File: src/ast/sentences/ConditionalSentence.java
package ast.sentences;

import ast.AbstractLocatable;
import ast.expressions.Expression;

public abstract class ConditionalSentence extends AbstractLocatable implements Sentence {

    private Expression condition;
    private Sentence body;

    public ConditionalSentence(Expression condition, Sentence body, int line, int column) {
        super(line, column);
        this.condition = condition;
        this.body = body;
    }

    public Expression getCondition() {
        return condition;
    }

    public Sentence getBody() {
        return body;
    }
}
